javascript/readme.md

```diff
@@ -1,3 +1,386 @@
+
+### Labels
+
+Statement labels should be avoided. Only these statements should be labeled: `while`, `do`, `for`, `switch`.
+
+### return Statement
+
+The return value expression must start on the same line as the return keyword in order to avoid semicolon insertion.
+
+### if Statement
+
+The if class of statements should have the following form:
+
+```javascript
+if (condition) {
+    statements
+}
+
+if (condition) {
+    statements
+} else {
+    statements
+}
+
+if (condition) {
+    statements
+} else if (condition) {
+    statements
+} else {
+    statements
+}
+```
+
+### for Statement
+
+A for class of statements should have the following form:
+
+```javascript
+for (initialization; condition; update) {
+    statements
+}
+```
+
+### while Statement
+
+A while statement should have the following form:
+
+```javascript
+while (condition) {
+    statements
+}
+```
+
+### do Statement
+
+A do statement should have the following form:
+
+```javascript
+do {
+    statements
+} while (condition);
+```
+
+Unlike the other compound statements, the do statement always ends with a ; semicolon.
+
+### switch Statement
+
+A switch statement should have the following form:
+
+```javascript
+switch (expression) {
+case expression:
+    statements
+default:
+    statements
+}
+```
+
+Each case is aligned with the switch. This avoids over-indentation. A case label is not a statement, and should not be indented like one.
+
+Each group of statements (except the default) should end with break, return, or throw. Do not fall through.
+
+### try Statement
+
+The try class of statements should have the following form:
+
+```javascript
+try {
+    statements
+} catch (variable) {
+    statements
+}
+
+try {
+    statements
+} catch (variable) {
+    statements
+} finally {
+    statements
+}
+```
+
+### continue Statement
+
+Avoid use of the continue statement. It tends to obscure the control flow of the function.
+
+### with Statement
+
+The with statement should not be used.
+
+## {} and []
+
+Use `{}` instead of new `Object()`. Use `[]` instead of `new Array()`.
+
+Use arrays when the member names would be sequential integers. Use objects when the member names are arbitrary strings or names.
+
+### , comma Operator
+
+Avoid the use of the comma operator. (This does not apply to the comma separator, which is used in object literals, array literals, var statements, and parameter lists.)
+
+### Assignment Expressions
+
+Avoid doing assignments in the condition part of if and while statements.
+
+Is
+
+```javascript
+if (a = b) {
+```
+
+a correct statement? Or was
+
+```javascript
+if (a == b) {
+```
+
+intended? Avoid constructs that cannot easily be determined to be correct.
+
+### === and !== Operators.
+
+Use the `===` and `!==` operators. The `==` and `!=` operators do type coercion and should not be used.
+
```

Comment:
One other thing that I would like to be added in, is testing for 'undefined' should be if(undefined == variable).